frontend/src/views/login/login.tsx

```diff
@@ -0,0 +1,20 @@
+import { Button } from '@mui/material'
+import './login.scss'
+import { useCallback } from 'react'
+import { useAuth } from '../../context/authContext'
+
+export function Login() {
+  const user = useAuth()
+
+  const handleLogin = useCallback(() => {
+    if (!user) {
+      throw new Error('useAuth must be used within an AuthProvider')
+    }
```

Comment:
I believe it would throw an error if `useAuth` is not used within an `AuthProvider` (i.e. error thrown at line 7), so no need for this line?

Reply:
That was necessary since typescript enforces that user: (AuthContextType | null) can't call its methods unless
null check is handled.

You have a point that an error is already thrown. I have thus replaced the explicit error throwing with the non-null assertion operator appropriately.

Reply:
I see, normally I would use typecast instead. But this also works.